### DISCUSSION TITLE
Override Rates

### DISCUSSION THREAD

**Discussion by psfshr** [BottlecapDave/HomeAssistant-OctopusEnergy#841] [Q&A] [answered]
So as I've noted in #839 I'm playing with the override rates to try and compare what I'd pay on Agile vs my current Flex tariff.
I turned on the override entities yesterday and put in (what would be) my Agile tariff (E-1R-AGILE-24-04-03-H) and it duly pulled down the previous day's rates which I could then graph against by current tariff (working around #839).

Should those override tariffs now be updating on a daily basis (e.g. event.octopus_energy_electricity_xxx_yyy_previous_consumption_override_rates), as currently they're still showing data for the 14th/15th and nothing for the 16th (it's currently 16:31 on the 17th)?

The "state" on the event is for today (2024-04-17T15:07:52.167+00:00) which suggests it's doing something, and manually querying the API via CURL returns rates for today.

Edit: The normal current and previous day events seem to populate correctly with today's and yesterday's rates as expected.

**Comment by BottlecapDave** [comment, marked as answer]
Hello. The octopus_energy_electricity_xxx_yyy_previous_consumption_override_rates is raised everytime the rate information is retrieved, however it retrieves it for the same period as the previous consumption data. Therefore if the previous consumption data doesn't update (e.g. because it takes longer than the default 24 hours) then the rate information won't update. Is the previous consumption data updating/available?

**Comment by psfshr** [comment]
So I think that's the answer. Looking at what is updating and when it does seem to be updating in line with the previous day consumption, but that previous consumption data isn't updating for me until later in the day (typical OE). When I was setting up the overrides and testing, I was typically checking in the morning, but checking in late afternoon/evening it's then there.
